packages/medusa/src/types/common.ts

```diff
@@ -16,17 +18,33 @@ export interface FindConfig<Entity> {
 export type PaginatedResponse = { limit: number; offset: number; count: number }
 
 export class DateComparisonOperator {
+  @IsOptional()
   @IsDate()
-  lt?: Date | string
+  @Transform(({ value }) =>
+    !isNaN(Date.parse(value)) ? new Date(value) : new Date(Number(value) * 1000)
+  )
+  lt?: Date
 
+  @IsOptional()
   @IsDate()
-  gt?: Date | string
+  @Transform(({ value }) =>
+    !isNaN(Date.parse(value)) ? new Date(value) : new Date(Number(value) * 1000)
+  )
+  gt?: Date
 
+  @IsOptional()
   @IsDate()
-  gte?: Date | string
+  @Transform(({ value }) =>
+    !isNaN(Date.parse(value)) ? new Date(value) : new Date(Number(value) * 1000)
+  )
+  gte?: Date
 
+  @IsOptional()
   @IsDate()
-  lte?: Date | string
+  @Transform(({ value }) =>
+    !isNaN(Date.parse(value)) ? new Date(value) : new Date(Number(value) * 1000)
+  )
+  lte?: Date
```

Comment:
Nit: could extract comparison function to reduce the size and repetitiveness of the `DateComparisonOperator`